ffi/derive/src/ffi_fn.rs

```diff
@@ -183,30 +198,31 @@ fn gen_method_call_stmt(fn_descriptor: &FnDescriptor) -> TokenStream {
 }
 
 fn gen_output_assignment_stmts(fn_descriptor: &FnDescriptor) -> TokenStream {
-    if let Some(output_arg) = &fn_descriptor.output_arg {
-        if let Some(receiver) = &fn_descriptor.receiver {
-            let arg_name = receiver.name();
-            let src_type = receiver.src_type();
-
-            if matches!(src_type, syn::Type::Path(_)) {
-                return quote! {
-                    if __out_ptr.is_null() {
-                        return Err(iroha_ffi::FfiReturn::ArgIsNull);
-                    }
-
-                    __out_ptr.write(#arg_name);
-                };
+    match &fn_descriptor.output_arg {
+        Some(output_arg) if is_fn_consume_and_return_self(fn_descriptor) => {
+            let arg_name = output_arg.name();
+            quote! {
+                if __out_ptr.is_null() {
+                    return Err(iroha_ffi::FfiReturn::ArgIsNull);
+                }
+
+                __out_ptr.write(#arg_name);
             }
         }
-
-        let (arg_name, arg_type) = (output_arg.name(), output_arg.ffi_type_resolved(true));
-        let output_arg_conversion = crate::util::gen_arg_src_to_ffi(output_arg, true);
-
-        return quote! {
-            #output_arg_conversion
-            iroha_ffi::OutPtrOf::<#arg_type>::write(__out_ptr, #arg_name)?;
-        };
+        Some(output_arg) => {
+            let (arg_name, arg_type) = (output_arg.name(), output_arg.ffi_type_resolved(true));
+            let output_arg_conversion = crate::util::gen_arg_src_to_ffi(output_arg, true);
+            quote! {
+                #output_arg_conversion
+                iroha_ffi::OutPtrOf::<#arg_type>::write(__out_ptr, #arg_name)?;
+            }
+        }
+        None => quote! {},
     }
+}
 
-    quote! {}
+/// Check if fn signature is like `fn(self, ...) -> Self`
```

Comment:
Honestly, this is a rare case of a function being named so-well that this doc is redundant.